server/src/main/java/org/elasticsearch/transport/TransportHandshaker.java

```diff
@@ -257,6 +262,19 @@ static void ensureCompatibleVersion(
         Object channel
     ) {
         if (TransportVersion.isCompatible(remoteTransportVersion)) {
+            if (remoteTransportVersion.before(V8_18_HANDSHAKE_VERSION)) {
+                deprecationLogger.warn(
+                    DeprecationCategory.OTHER,
+                    "handshake_version",
+                    "Performed a handshake with a remote node whose version will be incompatible after this node "
+                        + "is upgraded to 9.x. This is likely a result of a CCR/CCS request. "
+                        + "Remote version: [{}]. "
+                        + "This version: [{}]. "
+                        + "Ensure you upgrade the remote node to a version that is compatible with this node.",
+                    remoteTransportVersion.toReleaseVersion(),
+                    localTransportVersion.toReleaseVersion()
+                );
+            }
```

Comment:
I'm looking to ensure that:
- This warning is produced in the right location in the code.
- This is the right message